cloudsql/mysql/Startup.cs

```diff
@@ -0,0 +1,242 @@
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+using Google.Cloud.Diagnostics.AspNetCore;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using MySql.Data.MySqlClient;
+using Npgsql;
+using System;
+using System.Data.Common;
+using System.Data;
+using Polly;
+
+namespace CloudSql
+{
+    public class Startup
+    {
+        public IConfiguration Configuration { get; }
+        IServiceCollection _services;
+
+        public Startup(IConfiguration configuration)
+        {
+            Configuration = configuration;
+        }
+
+        // This method gets called by the runtime. Use this method to add
+        // services to the container.
+        // For more information on how to configure your application, visit
+        // http://go.microsoft.com/fwlink/?LinkID=398940
+        public void ConfigureServices(IServiceCollection services)
+        {
+            string projectId = Google.Api.Gax.Platform.Instance().ProjectId;
+            if (!string.IsNullOrEmpty(projectId))
+            {
+                services.AddGoogleExceptionLogging(options =>
+                {
+                    options.ProjectId = projectId;
+                    options.ServiceName = "CloudSqlSample";
+                    options.Version = "Test";
+                });
+            }
+            services.AddSingleton(typeof(DbConnection), (IServiceProvider) =>
+                InitializeDatabase());
+            services.AddMvc(options =>
+            {
+                options.Filters.Add(typeof(DbExceptionFilterAttribute));
+            });
+            _services = services;
+        }
+
+        DbConnection InitializeDatabase()
+        {
+            DbConnection connection;
+            string database = Configuration["CloudSQL:Database"];
+            switch (database.ToLower())
+            {
+                case "mysql":
+                    connection = GetMySqlConnection();
+                    break;
+                case "postgresql":
+                    connection = GetPostgreSqlConnection();
+                    break;
+                default:
+                    throw new ArgumentException(string.Format(
+                        "Invalid database {0}.  Fix appsettings.json.",
+                            database), "CloudSQL:Database");
+            }
+            connection.Open();
+            using (var createTableCommand = connection.CreateCommand())
+            {
+                createTableCommand.CommandText = @"
+                    CREATE TABLE IF NOT EXISTS
+                    votes(
+                        vote_id SERIAL NOT NULL,
+                        time_cast timestamp NOT NULL,
+                        candidate CHAR(6) NOT NULL,

```

Comment:
I (think) there was a small issue with char and varchar causing an issue between MySQL and Postgres. 

For [MySQL](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/6ec5bed56607605a443037c26e659c1f1282928e/cloud-sql/mysql/sqlalchemy/main.py#L92) we used CHAR
For [Postgres](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/cloud-sql/postgres/sqlalchemy/main.py#L92) we used VARCHAR